### PR TITLE
Dt 1649 call update api

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: server/routes/courtRegister/allCourtsViewMapper.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { AllCourts } from '../../services/courtRegisterService'
```
Comment (andymarke) [inline server/routes/courtRegister/allCourtsViewMapper.ts:1]: get rid of word Mapper

File: server/routes/courtRegister/courtRegisterController.test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import CourtRegisterService from '../../services/courtRegisterService'
+import CourtRegisterController from './courtRegisterController'
+
+jest.mock('../../services/courtRegisterService')
+
+describe('Court Register controller', () => {
+  let courtRegisterService: jest.Mocked<CourtRegisterService>
+  let controller: CourtRegisterController
+  const request = {} as Request
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (andymarke) [inline server/routes/courtRegister/courtRegisterController.test.ts:11]: is there a better way?

File: server/routes/testutils/appSetup.ts [hunk 3]
```diff
@@ -32,6 +34,18 @@ class MockUserService extends UserService {
   }
 }
 
+class MockCourtRegisterService extends CourtRegisterService {
```
Comment (andymarke) [inline server/routes/testutils/appSetup.ts:37]: extract to different file

File: feature.env [hunk 4]
```diff
@@ -1,6 +1,7 @@
 PORT=3007
 HMPPS_AUTH_URL=http://localhost:9091/auth
 TOKEN_VERIFICATION_API_URL=http://localhost:9091/verification
+COURT_REGISTER_API_URL=http://localhost:9091/court-register
```
Comment (andymarke) [inline feature.env:4]: need to add docker compose

**Review by mikehalmamoj — APPROVED** [review]

File: server/services/courtRegisterService.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import type HmppsAuthClient from '../data/hmppsAuthClient'
+import RestClient from '../data/restClient'
+import config from '../config'
+import logger from '../../logger'
+
+export interface Court {
```
Comment (andymarke) [inline server/services/courtRegisterService.ts:6]: import this from swagger

File: server/routes/courtRegister/courtRegisterController.test.ts (hunk 2 above)
Comment (andrewrlee) [reply server/routes/courtRegister/courtRegisterController.test.ts:11]: find it pretty painful to test these in mock tests. 

We do this in BVL:   
```
const res = ({
    locals: {},
    render: jest.fn(),
    redirect: jest.fn(),
  } as unknown) as jest.Mocked<Response>
```

Would probably just use superagent going forward

File: server/routes/testutils/appSetup.ts (hunk 3 above)
Comment (andrewrlee) [reply server/routes/testutils/appSetup.ts:37]: one of the nice things about using classes is that you can mock them quite nicely using jest:

https://github.com/ministryofjustice/hmpps-book-video-link/blob/main/backend/routes/amendBooking/changeDateAndTimeController.test.ts#L8-L13

We don't have a great way of injecting this in use of force

**Pull request merged**